src/ducks/pullrequests/index.js

```diff
@@ -1,153 +1,88 @@
-              computePullRequestTargetLink(
-                computePullRequestOriginLink(
-                  computePullRequestLink(pullrequest, fn2), fn1), fn1))))
-    )
-  )
-)
-
 
 /**
  * Initial state
  */
 const initialState = {
   error: null,
   isFetching: false,
-  byId: {},
-  allIds: [],
+  entities: {},
+  pagination: {
+    total: 0,
+    pages: {},
+    pageSize: 0,
+    currentPage: 0,
+  },
 }
 
-/**
- * Pull request `byId` state reducer
- */
-const entities = (state = {}, action) => {
-  switch (action.type) {
-    // case 'UPDATE_PULLREQUEST':
-    //   return {
-    //     ...state,
-    //     [action.id]: {
-    //       ...action.pullrequests
-    //     }
-    //   }
-    default:
-      if (action.pullrequests) {
-        return _.merge({}, state, action.pullrequests)
-      }
-      return state
-  }
+export type PullRequestDictionary = {
+  [id: string]: Object
 }
 
-/**
- * Pull request `allIds` state reducer
- */
-const ids = (state: Array<string> = [], action: Object) => {
-  switch (action.type) {
-    default:
-      if (action.ids) {
-        return state.concat(action.ids)
-      }
-      return state
-  }
+export type PullRequestsStateType = {
+  error: ?string,
+  isFetching: boolean,
+  entities: PullRequestDictionary,
 }
 
+export const entitiesReducer = combineReducers({
+  entities,
+  error,
+  isFetching,
+  pagination,
+})
+
 /**
  * Pullrequests reducer
  */
-export default (state: Object = initialState, action: Object): Object => {
+export default (
+  state: PullRequestsStateType = initialState, action: Object): PullRequestsStateType => {
   switch (action.type) {
     case types.SET_PULL_REQUESTS:
-      return {
-        ...state,
-        byId:
-          entities(state.byId, { pullrequests: reduceArrayToObj(action.pullrequests
-            .map(x =>
-              computePullRequest(x)(helpers.buildPullRequestLink)(helpers.buildProjectLink))) }),
-        allIds: ids(state.allIds, { ids: action.pullrequests.map(x => x.id) }),
-      }
-    case types.SENDING_REQUEST:
-      return {
-        ...state,
-        isFetching: action.sending,
-      }
-    case types.REQUEST_ERROR:
-      return {
-        ...state,
-        error: action.error,
-      }
-    case types.CLEAR_ERROR:
-      return {
-        ...state,
-        error: null,
-      }
+      return entitiesReducer(state, entitiesActions.setEntities(action.nodes))
+    case types.FETCH_PULL_REQUESTS:
+      return entitiesReducer(state, requestPage(action))
     default:
-      return state
+      return entitiesReducer(state, action)
   }
 }
 
 /**
  * Actions
  */
 export const actions = {
-  sendingRequest: (sending: boolean): Object => ({ type: types.SENDING_REQUEST, sending }),
-  requestError: (error: string): Object => ({ type: types.REQUEST_ERROR, error }),
-  clearError: (): Object => ({ type: types.CLEAR_ERROR }),
+  // sendingRequest: (sending: boolean) => ({ type: entitiesTypes.SENDING_REQUEST, sending }),
```

Comment:
Did we disable eslint for  `yarn run dev`, it stopped warning me about linting errors? But it did before. We need to enable it again because I might miss more eslint errors in the future :)